rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/filter/TbCheckAlarmStatusNode.java

```diff
@@ -0,0 +1,96 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.rule.engine.filter;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import lombok.extern.slf4j.Slf4j;
+import org.thingsboard.rule.engine.api.RuleNode;
+import org.thingsboard.rule.engine.api.TbContext;
+import org.thingsboard.rule.engine.api.TbNode;
+import org.thingsboard.rule.engine.api.TbNodeConfiguration;
+import org.thingsboard.rule.engine.api.TbNodeException;
+import org.thingsboard.rule.engine.api.util.TbNodeUtils;
+import org.thingsboard.server.common.data.alarm.Alarm;
+import org.thingsboard.server.common.data.plugin.ComponentType;
+import org.thingsboard.server.common.msg.TbMsg;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+
+import static org.thingsboard.rule.engine.api.TbRelationTypes.FAILURE;
+import static org.thingsboard.rule.engine.api.TbRelationTypes.SUCCESS;
+
+@Slf4j
+@RuleNode(
+        type = ComponentType.FILTER,
+        name = "checks alarm status",
+        configClazz = TbCheckAlarmStatusNodeConfig.class,
+        nodeDescription = "Checks alarm status.",
+        nodeDetails = "If the alarm status matches the specified one - msg is success if does not match - msg is failure.",
+        uiResources = {"static/rulenode/rulenode-core-config.js"},
+        configDirective = "tbFilterNodeCheckAlarmStatusConfig")
+public class TbCheckAlarmStatusNode implements TbNode {
+    private TbCheckAlarmStatusNodeConfig config;
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    @Override
+    public void init(TbContext tbContext, TbNodeConfiguration configuration) throws TbNodeException {
+        this.config = TbNodeUtils.convert(configuration, TbCheckAlarmStatusNodeConfig.class);
+    }
+
+    @Override
+    public void onMsg(TbContext ctx, TbMsg msg) throws TbNodeException {
+        try {
+            Alarm alarm = mapper.readValue(msg.getData(), Alarm.class);
+
+            ListenableFuture<Alarm> latest = ctx.getAlarmService().findAlarmByIdAsync(ctx.getTenantId(), alarm.getId());
+
+            Futures.addCallback(latest, new FutureCallback<Alarm>() {
+                @Override
+                public void onSuccess(@Nullable Alarm result) {
+                    boolean isPresent = false;
+                    for (String alarmStatus : config.getAlarmStatusList()) {
+                        if (alarm.getStatus().name().equals(alarmStatus)) {
+                            isPresent = true;
+                            break;
+                        }
+                    }
+
+                    if (isPresent) {
+                        ctx.tellNext(msg, SUCCESS);
```

Comment:
Please use "True"  and "False" in case alarm is present. Please use Failure in case no such alarmId found or failure of DB query, etc.